go.mod

```diff
@@ -1,16 +1,19 @@
 module github.com/kcp-dev/code-generator/v2
 
-go 1.22.0
+go 1.23.0
+
+toolchain go1.23.4
```

Comment:
Let's remove this `toolchain` directive. I know go tooling adds it automatically, but it enforces usage of 1.23.4, I don't think we really need that. I quickly checked upstream `go.mod` files and they don't use it either.